Sources/Clappr_iOS/Classes/Animation/DoubleTapAnimation.swift

```diff
@@ -0,0 +1,254 @@
+                bubbleView.alpha = 0.0
+                self.core?.view.layoutSubviews()
+            }, completion: { _ in
+                heightConstraint.constant = 0
+                widthConstraint.constant = 0
+                self.core?.view.layoutSubviews()
+            })
+        })
+    }
+    
+    private func animate(_ label: UILabel) {
+        core?.view.bringSubview(toFront: label)
+        UIView.animate(withDuration: 0.2, animations: {
+            label.alpha = 1.0
+            self.core?.view.layoutSubviews()
+        }, completion: { _ in
+            UIView.animate(withDuration: 0.2, delay: 0.5, animations: {
+                label.alpha = 0.0
+                self.core?.view.layoutSubviews()
+            })
+        })
+    }
+    
+    private func animate(_ image: UIImageView, delay: TimeInterval) {
+        core?.view.bringSubview(toFront: image)
+        UIView.animate(withDuration: 0.15, delay: delay, animations: {
+            image.alpha = 1.0
+            self.core?.view.layoutSubviews()
+        }, completion: { _ in
+            UIView.animate(withDuration: 0.15, delay: 0.15, animations: {
+                image.alpha = 0.0
+                self.core?.view.layoutSubviews()
+            })
+        })
+    }
+    
+    private func setup(_ core: Core?) {
+        guard let view = core?.view else { return }
+        
+        view.clipsToBounds = true
+        
+        setupBubble(leftBubbleView, within: view, heightConstraint: &leftBubbleHeight, widthConstraint: &leftBubbleWidth, position: .leading)
+        setupBubble(rightBubbleView, within: view, heightConstraint: &rightBubbleHeight, widthConstraint: &rightBubbleWidth, position: .trailing)
+        
+        setupLabel(view, label: fowardLabel, position: 1.5)
+        setupLabel(view, label: backLabel, position: 0.5)
+        
+        setupImage(view, image: fowardIcon1, label: fowardLabel, constX: -14)
+        setupImage(view, image: fowardIcon2, label: fowardLabel, constX: 0)
+        setupImage(view, image: fowardIcon3, label: fowardLabel, constX: 14)
+        
+        setupImage(view, image: backIcon1, label: backLabel, constX: -14)
+        setupImage(view, image: backIcon2, label: backLabel, constX: 0)
+        setupImage(view, image: backIcon3, label: backLabel, constX: 14)
+    }
+    
+    private func mirrorImage(_ view: UIImageView) -> UIImageView {
+        view.transform = CGAffineTransform(rotationAngle: (180.0 * .pi) / 180.0)
+        return view
+    }
+    
+    private func setupBubble(_ bubble: UIView, within view: UIView, heightConstraint: inout NSLayoutConstraint, widthConstraint: inout NSLayoutConstraint, position: NSLayoutAttribute) {
+        
+        bubble.backgroundColor = UIColor(white: 0, alpha: 0.2)
+        view.addSubview(bubble)
+        bubble.translatesAutoresizingMaskIntoConstraints = false
+        
+        view.addConstraint(NSLayoutConstraint(item: bubble,
```

Comment:
There's always the endless discussion about viewcode or xibs. Maybe in this case we could have reduced the amount of code by using a nib.

We do not need to do that now but it's a discussion that we can have later.